### PR TITLE
[JUJU-712] Add the snapcraft plugs needed for strictly confining the juju snap

### PR DESCRIPTION
**Pull request by wallyworld**
Add changes to `snapcraft.yaml` to enable a strictly confined snap to be built.

## QA steps

`snapcraft remote-build --launchpad-accept-public-upload --build-on amd64`

### CODE REVIEW

File: provider/lxd/provider.go [hunk 1]
```diff
@@ -221,7 +222,9 @@ func (p *environProvider) DetectClouds() ([]cloud.Cloud, error) {
 		configPath := filepath.Join(dir, "config.yml")
 		config, err := p.lxcConfigReader.ReadConfig(configPath)
 		if err != nil {
-			logger.Warningf("unable to read/parse LXC config file (%s): %s", configPath, err)
+			if !strings.HasSuffix(err.Error(), "permission denied") {
```
Comment (ycliuhw) [inline provider/lxd/provider.go:225]: Do we want to ignore the "permission denied" error?

**Review by ycliuhw — APPROVED** [review]
LGTM ty

File: provider/lxd/provider.go (hunk 1 above)
Comment (wallyworld) [reply provider/lxd/provider.go:225]: yeah cause it only happens when using a strict snap and from what i can tell, the dir under snap/lxd/common is preferred anyway

**Comment by wallyworld** [comment]
$$merge$$

**Pull request merged**